tempesta_fw/http.c

```diff
@@ -1281,7 +1296,7 @@ tfw_http_conn_resend(TfwSrvConn *srv_conn, bool first, struct list_head *eq)
 	{
 		if (tfw_http_req_evict(srv_conn, srv, req, eq))
 			continue;
-		if (!tfw_http_req_fwd_send(srv_conn, srv, req, eq))
+		if (tfw_http_req_fwd_send(srv_conn, srv, req, eq, true))
```

Comment:
By the way, why `tfw_http_conn_resend()` and `tfw_http_conn_fwd_unsent()` are so different? Seems like both should has the same role: send requests one-by-one from the queue until error happen. As for me `tfw_http_conn_resend()` should also be able to stop at the first error. It's also affected by the issue.

Reply:
This place confuses me. Suppose there is a first retry to resend a request while we have say 5 retries in configuration. This try fails. `tfw_http_req_fwd_send()` because of `drain == true` adds the request to error queue `eq` and next we send an error message to a user. Why? For me it seems that we have the same scenario which the patch tries to fix. If `tfw_http_req_evict()` didn't "eat" the request then probably it has many tries...

Reply:
Added storing and rescheduling for requests re-sending procedure.